internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -739,6 +764,42 @@ func (r *BMACReconciler) ensureSpokeMachine(ctx context.Context, spokeClient cli
 	return machineSpoke, nil
 }
 
+// Get MCS cert, encode it and add it to BMH which will add ultimately into agent
+func (r *BMACReconciler) ensureMCSCert(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost) error {
+	configMap := &corev1.ConfigMap{}
+	key := types.NamespacedName{
+		Namespace: "kube-system",
+		Name:      "root-ca",
+	}
+	if err := r.Get(ctx, key, configMap); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return err
+		}
+	}
+	if configMap.Data == nil {
+		return errors.Errorf("Configmap %s/%s  does not contain any data", configMap.Namespace, configMap.Name)
+	}
+	certData, ok := configMap.Data[MCS_CERT_NAME]
+	if !ok || len(certData) == 0 {
+		return errors.Errorf("Configmap data for %s/%s  does not contain %s", configMap.Namespace, configMap.Name, MCS_CERT_NAME)
+	}
+	mcsCrt := configMap.Data[MCS_CERT_NAME]
+	encocedMCSCrt := base64.StdEncoding.EncodeToString([]byte(mcsCrt))
+
+	ignitionWithMCSCert, err := r.formatSecondDayWorkerIgnition(encocedMCSCrt)
+	if err != nil {
+		return errors.Errorf("Failed to create ignition string with MCS cert")
+	}
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
+	if _, ok := bmhAnnotations[BMH_AGENT_IGNITION_CONFIG_OVERRIDES]; !ok {
+		if bmh.ObjectMeta.Annotations == nil {
+			bmh.ObjectMeta.Annotations = make(map[string]string)
+		}
+		bmh.ObjectMeta.Annotations[BMH_AGENT_IGNITION_CONFIG_OVERRIDES] = ignitionWithMCSCert
+	}
```

Comment:
I have a slightly different approach.
1. If the user has not provided ignition override via annotation, then I am adding the annotation with the new mcsignition containing just the version and security.tls.certificateAuthorities in it
https://github.com/pawanpinjarkar/assisted-service/blob/MGMT-4561/internal/controller/controllers/bmh_agent_controller.go#L791-L794
2. If the user has provided ignition override via annotation, but whether it contains the security.tls.certificateAuthorities or not I am always merging the new mcsignition containing just the version and security.tls.certificateAuthorities into the user-provided ignition. WDYT about this approach?

Note: In this case, there is an issue in the test case that in case 2 above, the user-provided ignition is getting rejected during `ParseToLatest()` although the format of user-provided ignition is right. Something to look more into.